jobs/mongodb_migration/src/mongodb_migration/migrations/_20231106193200_cache_add_partial_field_in_split_duckdb_index.py

```diff
@@ -0,0 +1,60 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2023 The HuggingFace Authors.
+
+import logging
+
+from libcommon.constants import CACHE_COLLECTION_RESPONSES, CACHE_MONGOENGINE_ALIAS
+from libcommon.simple_cache import CachedResponseDocument
+from mongoengine.connection import get_db
+
+from mongodb_migration.check import check_documents
+from mongodb_migration.migration import Migration
+
+
+# connection already occurred in the main.py (caveat: we use globals)
+class MigrationAddPartialToSplitDuckdbIndexCacheResponse(Migration):
+    def up(self) -> None:
+        # See https://docs.mongoengine.org/guide/migration.html#example-1-addition-of-a-field
+        logging.info(
+            "If missing, add the 'partial', 'num_rows' and 'num_bytes' fields with the default value"
+            " (False, None, None) to the cached results of split-duckdb-index"
+        )
+        db = get_db(CACHE_MONGOENGINE_ALIAS)
+        db[CACHE_COLLECTION_RESPONSES].update_many(
+            {
+                "kind": "split-duckdb-index",
+                "http_status": 200,
+                "content.partial": {"$exists": False},
+            },
+            {
+                "$set": {
+                    "content.partial": True,
```

Comment:
isn't it False?

Reply:
Good catch ! And actually it should be `None` since it may have indexed a partial parquet dump so we have to keep the `Optional`